taichi/ui/common/gui_base.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include <string>
+#include "taichi/ui/utils/utils.h"
+
+TI_UI_NAMESPACE_BEGIN
+
+class GuiBase {
+ public:
+  virtual void begin(std::string name,
```

Comment:
Use virtual ...() = 0;?
You need to add a virtual destructor, e.g. virtual ~CanvasBase() = default;